ext/handlers_http.h

```diff
@@ -201,7 +201,79 @@ static inline zend_string *ddtrace_serialize_baggage(HashTable *baggage) {
     return serialized_baggage.s;
 }
 
-static inline void ddtrace_inject_distributed_headers_config(zend_array *array, bool key_value_pairs, zend_array *inject) {
+typedef enum {
+    HEADER_MODE_ARRAY,       // ["Header: value", ...]
+    HEADER_MODE_KV_PAIRS,    // [key => value]
+    HEADER_MODE_CONTEXT      // "Header1: value\r\nHeader2: value"
+} header_format;
+
+static inline zend_string *ddtrace_extract_header_key(zval *header) {
+    if (Z_TYPE_P(header) != IS_STRING) {
+        return NULL;
+    }
+
+    const char *val = Z_STRVAL_P(header);
+    const char *colon = strchr(val, ':');
+    if (!colon) {
+        return NULL;
+    }
+
+    size_t key_len = (size_t)(colon - val);
+    zend_string *key = zend_string_init(val, key_len, 0);
+    zend_str_tolower(ZSTR_VAL(key), key_len);
+    return key;
+}
+
+static inline HashTable *ddtrace_init_header_keys_set(zend_array *array, header_format format) {
+    HashTable *set = zend_new_array(0);
+
+    if (format != HEADER_MODE_ARRAY) {
+        return set;
+    }
+
+    ulong idx = 0;
+    ZEND_HASH_FOREACH_VAL(array, zval *entry) {
+        zend_string *header_key = ddtrace_extract_header_key(entry);
+        if (!header_key) {
+            idx++;
+            continue;
+        }
+
+        zval index_zv;
+        ZVAL_LONG(&index_zv, idx);
+        zend_hash_update(set, header_key, &index_zv);
+        zend_string_release(header_key);
+        idx++;
+    } ZEND_HASH_FOREACH_END();
+
+    return set;
+}
+
+static inline void ddtrace_add_or_update_header_array(zend_array *array, HashTable *keys, const char *header_key, size_t header_key_len, zend_string *header) {
+    zend_string *key = zend_string_init(header_key, header_key_len, 0);
+    zend_str_tolower(ZSTR_VAL(key), header_key_len);
+
+    // Check for duplicates
+    zval *existing_idx_zv = zend_hash_find(keys, key);
+    if (existing_idx_zv && Z_TYPE_P(existing_idx_zv) == IS_LONG) {
+        zend_hash_index_del(array, Z_LVAL_P(existing_idx_zv));
+    }
+
+    // Update keys set
+    zend_long new_idx = zend_hash_num_elements(array);
+    zval idx_zv;
+    ZVAL_LONG(&idx_zv, new_idx);
+    zend_hash_update(keys, key, &idx_zv);
```

Comment:
you may wrap this in `if (zend_hash_num_elements(keys) > 0)` to avoid some computations in a common case without pre-existing headers